File: ecos-records/src/main/java/ru/citeck/ecos/records2/meta/attproc/AttPrefixSuffixProcessor.java
package ru.citeck.ecos.records2.meta.attproc;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.commons.data.DataValue;
import ru.citeck.ecos.records2.RecordMeta;

import java.util.List;

public class AttPrefixSuffixProcessor implements AttProcessor {

    @NotNull
    @Override
    public Object process(@NotNull RecordMeta meta, @NotNull DataValue value, @NotNull List<DataValue> arguments) {

        String prefix = "";
        String suffix = "";

        if (arguments.size() > 0) {
            prefix = arguments.get(0).asText();
        }
        if (arguments.size() > 1) {
            suffix = arguments.get(1).asText();
        }

        return prefix + value.asText() + suffix;
    }

    @Override
    public String getType() {
        return "presuf";
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/proc/AttPrefixSuffixProcessor.kt
package ru.citeck.ecos.records3.record.atts.proc

import ru.citeck.ecos.commons.data.DataValue
import ru.citeck.ecos.commons.data.ObjectData

class AttPrefixSuffixProcessor : AbstractAttProcessor<AttPrefixSuffixProcessor.Args>() {

    override fun processOne(attributes: ObjectData, value: DataValue, args: Args): Any? {
        val text = value.asText()
        return if (text.isEmpty()) {
            value
        } else {
            args.prefix + value.asText() + args.suffix
        }
    }

    override fun parseArgs(args: List<DataValue>): Args {
        var prefix = ""
        var suffix = ""
        if (args.isNotEmpty()) {
            prefix = args[0].asText()
        }
        if (args.size > 1) {
            suffix = args[1].asText()
        }
        return Args(prefix, suffix)
    }

    override fun getType(): String = "presuf"

    class Args(val prefix: String, val suffix: String)
}
